trunner/tools/phoenix.py

```diff
@@ -22,6 +22,30 @@ def wait_for_dev(port, timeout=0):
             raise TimeoutError
 
 
+def wait_for_vid_pid(port_hint: str, timeout=0):
+    """Searcher for connected usb device via vendor & product id"""
+
+    asleep = 0
+    port = None
+
+    while port is None:
+        time.sleep(0.01)
+        asleep += 0.01
+
```

Comment:
i would go for specifying pid/vid as a number and than:

```python
found_ports = []

while not found_ports:
  time.sleep(0.01)
  asleep += 0.01

  found_ports = [ port for port in list_ports.comports() if port.pid == pid and port.vid == vid ]
  if len(found_ports) > 1:
     raise Exception(...)

  # timeout handling here

return found_ports[0].device
```
(the code was not tested, I'm just presenting a more robust approach)